java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/launch/NbLaunchDelegate.java

```diff
@@ -344,19 +338,54 @@ public void propertyChange(PropertyChangeEvent evt) {
                     notifyFinished(context, exitCode != null && exitCode == 0);
                 });
                 Lookups.executeWith(execLookup, () -> {
-                    execNative(nativeImageFile, args, context, ed, launchFuture);
+                    execNative(nativeImageFile, args, context, ed, params, launchFuture);
                 });
             }
         }
         return launchFuture;
     }
 
-    private static void execNative(File nativeImageFile, List<String> args, DebugAdapterContext context, ExecutionDescriptor executionDescriptor, CompletableFuture<Void> launchFuture) {
+    private static ExplicitProcessParameters createExplicitProcessParameters(Map<String, Object> launchArguments) {
+        List<String> args = argsToStringList(launchArguments.get("args"));
+        List<String> vmArgs = argsToStringList(launchArguments.get("vmArgs"));
+        String cwd = Objects.toString(launchArguments.get("cwd"), System.getProperty("user.dir"));
```

Comment:
Q: is home dir sensible default -- shouldn't we use e.g. project base dir or something like that ?

Reply:
"user.dir" is not "user.home". CPP debugger was using "user.dir" unless set otherwise. I've corrected it not to set the working dir when no working dir is specified.